kolibri/core/public/utils.py

```diff
@@ -171,35 +210,51 @@ def request_soud_sync(server, user=None, queue_id=None, ttl=10):
         )
         return
 
+    if response.status_code == status.HTTP_404_NOT_FOUND:
+        return  # Request done to a server not owning this user's data
+
+    if response.status_code == status.HTTP_200_OK:
+        handle_server_sync_response(response, server, user)
+
+
+def handle_server_sync_response(response, server, user):
+    # In either case, we set the sync status for this user as queued
+    # Once the sync starts, then this should get cleared and the SyncSession
+    # set on the status, so that more info can be garnered.
     response_content = (
         response.content.decode()
         if isinstance(response.content, bytes)
         else response.content
     )
     server_response = json.loads(response_content or "{}")
-    if response.status_code == status.HTTP_404_NOT_FOUND:
-        return  # Request done to a server not owning this user's data
-
-    if response.status_code == status.HTTP_200_OK:
-        # In either case, we set the sync status for this user as queued
-        # Once the sync starts, then this should get cleared and the SyncSession
-        # set on the status, so that more info can be garnered.
-        UserSyncStatus.objects.update_or_create(user_id=user, defaults={"queued": True})
-
-        if server_response["action"] == SYNC:
-            job_id = startpeerusersync(server, user)
-            logger.info(
-                "Enqueuing a sync task for user {} in job {}".format(user, job_id)
+    UserSyncStatus.objects.update_or_create(user_id=user, defaults={"queued": True})
+
+    if server_response["action"] == SYNC:
+        SYNC_INTERVAL = OPTIONS["SYNCING"]["SYNC_INTERVAL"]
```

Comment:
Hrm, it might be simpler to just always read this from the server response for now. Because the options file only gets read at startup, writing to the options file doesn't make any guarantees about this option being updated, so this feels a little bit brittle potentially? In practice it may not be an issue for apps because everything will be running in a single process, so the in memory update should work.

Guessing that the reason to want to persist it into options is for the rescheduling of the sync? This might be a place where having a sync interval argument for the sync command might be helpful, as if it's not None then that's an indicator we should be syncing again, and also means we don't have to rely on this to persist the desired sync interval.

(One edge case that comes to mind is if two different servers with different sync intervals are both syncing with a SoUD, then these values could collide with each other).

Reply:
I had to modify the `sync` command and the functions calling it to accept this new interval parameter that's now sent by the server everytime a sync is accepted.